### PR TITLE
[mirror] Extract mirror packages to libmirror pkg

### PR DESCRIPTION
**Pull request by mvasl**
Extracted parts of mirror into publicly available library under `pkg/libmirror`

### CODE REVIEW

File: internal/mirror/cmd/modules/pull/pull.go [hunk 1]
```diff
@@ -101,18 +110,18 @@ func pullExternalModulesToLocalFS(
 
 	modulesFromRepo, err := modules.GetExternalModulesFromRepo(src.Spec.Registry.Repo, authProvider, insecure, skipVerifyTLS)
 	if err != nil {
-		return fmt.Errorf("Get external modules from %q: %w", src.Spec.Registry.Repo, err)
+		return fmt.Errorf("GetMinimalVersion external modules from %q: %w", src.Spec.Registry.Repo, err)
```
Comment (mvasl) [inline internal/mirror/cmd/modules/pull/pull.go:113]: goland autorefactoring crapped itself

File: internal/mirror/cmd/pull/pull.go [hunk 2]
```diff
@@ -253,7 +262,8 @@ func computeGOSTDigest(mirrorCtx *contexts.BaseContext) error {
 	if err = os.WriteFile(mirrorCtx.BundlePath+".gostsum", []byte(gostDigest), 0o666); err != nil {
 		return fmt.Errorf("Write GOST Checksum: %w", err)
 	}
-	log.InfoF("Digest: %s\nWritten to %s\n", gostDigest, mirrorCtx.BundlePath+".gostsum")
+
+	mirrorCtx.Logger.InfoF("Digest: %s\nWritten to %s\n", gostDigest, mirrorCtx.BundlePath+".gostsum")
```
Comment (mvasl) [inline internal/mirror/cmd/pull/pull.go:266]: Split into 2 InfoF calls

File: internal/mirror/cmd/pull/pull.go [hunk 3]
```diff
@@ -316,46 +327,46 @@ func PullDeckhouseToLocalFS(
...
+	logger.InfoLn("✅")
 
 	if err = layouts.PullDeckhouseReleaseChannels(pullCtx, imageLayouts); err != nil {
 		return fmt.Errorf("pull release channels: %w", err)
 	}
 
 	// We should not generate deckhousereleases.yaml manifest for single-release bundles
 	if pullCtx.SpecificVersion == nil {
-		log.InfoF("Generating DeckhouseRelease manifests...\t")
+		logger.InfoF("Generating DeckhouseRelease manifests...\t")
```
Comment (mvasl) [inline internal/mirror/cmd/pull/pull.go:346]: `...\t`

File: internal/mirror/cmd/vulndb/pull/pull.go [hunk 4]
```diff
@@ -18,15 +18,17 @@ package pull
...
 	"path/filepath"
 
 	"github.com/google/go-containerregistry/pkg/authn"
 	"github.com/spf13/cobra"
 	"k8s.io/component-base/logs"
 	"k8s.io/kubectl/pkg/util/templates"
 
-	"github.com/deckhouse/deckhouse-cli/internal/mirror/contexts"
-	"github.com/deckhouse/deckhouse-cli/internal/mirror/layouts"
+	contexts2 "github.com/deckhouse/deckhouse-cli/pkg/libmirror/contexts"
```
Comment (mvasl) [inline internal/mirror/cmd/vulndb/pull/pull.go:29]: 2

File: pkg/libmirror/layouts/push.go [hunk 5]
```diff
@@ -24,14 +24,15 @@ import (
 	"github.com/google/go-containerregistry/pkg/v1/layout"
 	"github.com/google/go-containerregistry/pkg/v1/remote"
 
-	mirror "github.com/deckhouse/deckhouse-cli/internal/mirror/util/auth"
-	"github.com/deckhouse/deckhouse-cli/internal/mirror/util/log"
+	"github.com/deckhouse/deckhouse-cli/pkg/libmirror/contexts"
+	mirror "github.com/deckhouse/deckhouse-cli/pkg/libmirror/util/auth"
```
Comment (mvasl) [inline pkg/libmirror/layouts/push.go:28]: auth

**Review by vadimartynov — APPROVED** [review]

**Pull request merged**